apps/microtvm/zephyr/template_project/src/mlperftiny/platform.cc

```diff
@@ -162,3 +172,33 @@ int8_t QuantizeFloatToInt8(float value, float scale, int zero_point) {
   }
   return (int8_t)(result);
 }
+
+// UART read.
+char TVMPlatformUartRxRead() {
```

Comment:
why some functions in this file have the `__attribute__((weak))` and some don't? What is the logic behind applying it?

Reply:
file is restructured, now all of them are with weak attribute.